Perf/range

KKould opened the pull request:
### What problem does this PR solve?

Non-persistent Level 0 to reduce the hard disk read and write latency and compaction impact caused by Level 0
- Use `BTreeTable` replace `SkipTable`

In order to improve iteration performance and avoid the performance loss of multiple levels of iteration caused by iteration on the hard disk side, the number of Level layers is reduced (7 -> 4), and the threshold for triggering `Sized Compaction` for each level is also reduced(10 -> 4).

details: https://www.notion.so/KipDB-3f456f92e559458ca54904bd04376ca6

### What is changed and how it works?

### Code changes

- [x] Has Rust code change
- [ ] Has CI related scripts change

### Check List

Tests <!-- At least one of them must be included. -->

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [x] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Note for reviewer

lewiszlw commented:
Do you have benchmark about skiplist and btree? And can we keep both (behind a feature flag) and let user to choose?

KKould commented:
> Do you have benchmark about skiplist and btree? And can we keep both (behind a feature flag) and let user to choose?

The reason for using BTreeTable to replace SkipTable is actually not that SkipList's performance is weak, but because Table needs to implement the `iter` API while ensuring Sync + Send, and SkipList has no official response to this.

issue: https://github.com/JP-Ellis/rust-skiplist/issues/24

Therefore I always comment out the code of SkipTableIter

But since the official did not respond for a long time, I switched to BTree Table and removed the SkipTable
